src/server/modemwatcher.cpp

```diff
@@ -31,15 +33,37 @@ ModemWatcher::ModemWatcher(const QString objectPath, const QString interface, QO
     QObject(parent), m_objectPath(objectPath), m_interface(interface), m_interfaceAvailable(false)
 {
     QDBusConnection::systemBus().connect(OfonoConstants::OFONO_SERVICE, m_objectPath,
-                                         OfonoConstants::OFONO_MODEM_INTERFACE, "PropertyChanged",
-                                         this, SLOT(onModemPropertyChanged(QString, QDBusVariant)));
+                                         OfonoConstants::OFONO_MODEM_INTERFACE,
+                                         "PropertyChanged", this,
+                                         SLOT(onModemPropertyChanged(QString, QDBusVariant)));
 
+    m_ofonoWatcher = new QDBusServiceWatcher(OfonoConstants::OFONO_SERVICE,
+                                             QDBusConnection::systemBus(),
+                                             QDBusServiceWatcher::WatchForRegistration,
+                                             this);
+    connect(m_ofonoWatcher, SIGNAL(serviceRegistered(QString)),
+            this, SLOT(getProperties()));
+
+    if (QDBusConnection::systemBus().interface()->isServiceRegistered(OfonoConstants::OFONO_SERVICE))
```

Comment:
This and similar checks are still synchronous dbus method calls. To dbus daemon itself, so they are unlikely to block for long, but still - any chance that we could track service availability in fully asynchronous manner?